tests/unit/modules/test_zypperpkg.py

```diff
@@ -739,6 +756,26 @@ def test_upgrade_success(self):
                         "--no-allow-vendor-change",
                     )
 
+                with patch(
+                    "salt.modules.zypperpkg.list_pkgs",
+                    MagicMock(side_effect=[{"vim": "1.1"}, {"vim": "1.2"}]),
+                ):
+                    ret = zypper.upgrade(
+                        dist_upgrade=True,
+                        fromrepo=["Dummy", "Dummy2"],
+                        novendorchange=False,
+                    )
+                    self.assertDictEqual(ret, {"vim": {"old": "1.1", "new": "1.2"}})
+                    zypper_mock.assert_any_call(
+                        "dist-upgrade",
+                        "--auto-agree-with-licenses",
+                        "--from",
+                        "Dummy",
+                        "--from",
+                        "Dummy2",
+                        "--allow-vendor-change",
```

Comment:
Is it a logic failure as you use `--allow-vendor-change` and `novendorchange=False` in this case or at least naming is just too confusing for me.

Reply:
Yeah that always bothered me.  I think, it'd be clearer to have a "allowvendorchange: true/false."  But I wasn't sure since it's been that way for years.

Reply:
@krionbsd but changing the option is also bad as it destroy existing states which use the old option already.
We should take care of backward compatibility.

There would be an option to introduce `allowvendorchange` in addition to `noallowvendorchange`.
The new var would overwrite the old one, in case it is specified.
The code would become more complex, but it would keep backward compatibility.